manager/allocator/networkallocator/networkallocator.go

```diff
@@ -157,6 +157,18 @@ func (na *NetworkAllocator) ServiceAllocate(s *api.Service) (err error) {
 	}
 
 	for _, nAttach := range s.Spec.Networks {
+		var vipAllocated bool
+		for _, vip := range s.Endpoint.VirtualIPs {
+			if vip.NetworkID == nAttach.Target {
+				vipAllocated = true
```

Comment:
Use a label here.

```go
outer:
  for {
    for {
      continue outer
    }
  }
```

This avoids the extra bool comparison and hard to read code.